File: Programmers/Level1/220819/javascript/CaesarCipher.test.js
import { solution } from './CaesarCipher';

test('lowerCase', () => {
  expect(solution('bc', 2)).toBe('de');
  expect(solution('z', 1)).toBe('a');
  expect(solution('zyx', 3)).toBe('cba');
});

test('lowerCaseWithBlank', () => {
  expect(solution('b c', 2)).toBe('d e');
});

test('upperCase', () => {
  expect(solution('AB', 1)).toBe('BC');
});

File: Programmers/Level1/220819/java/CaesarCipherTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class CaesarCipherTest {
  @Test
  void lower() {
    CaesarCipher caesarCipher = new CaesarCipher();

    assertEquals("def", caesarCipher.solution("abc", 3));
    assertEquals("a", caesarCipher.solution("z", 1));
  }

  @Test
  void upper() {
    CaesarCipher caesarCipher = new CaesarCipher();

    assertEquals("BC", caesarCipher.solution("AB", 1));
  }

  @Test
  void upperWithBlank() {
    CaesarCipher caesarCipher = new CaesarCipher();

    assertEquals("H L", caesarCipher.solution("F J", 2));
  }
}
